ui/pages/Login/components/Login.jsx

```diff
@@ -35,7 +36,7 @@ const Login = ({ onSubmit, googleLoginEnabled, location }) =>
       <Message visible warning>
         No existing seqr account for the Google user.
       </Message> }
-      <a href="/login/google-oauth2">Sign in with Google</a>
+      <a href={'/login/google-oauth2?next='.concat(`${queryString.parse(location.search).next || '/'}`)}>Sign in with Google</a>
```

Comment:
I updated this to fix the issue of redirecting to the next page when logging in with social auth.

Reply:
you shouldn't provide any empty next param, if there isn't one. Also, theres no harm in proxying all the query parameters, so it would be a lot easier to just use the whole unparsed `location.search`:
```
`/login/google-oauth2${location.search}`
```